File: MB-Secretaria/providers/provider-registry.ts
import type { ProviderAdapter, ProviderName } from "./types";

export class ProviderRegistry {
  private readonly providers = new Map<ProviderName, ProviderAdapter>();

  register(provider: ProviderAdapter): void {
    this.providers.set(provider.provider_name, provider);
  }

  get(name: ProviderName): ProviderAdapter | undefined {
    return this.providers.get(name);
  }

  list(): ProviderAdapter[] {
    return [...this.providers.values()];
  }
}

File: MB-Whatsapp/providers_py/provider_registry.py
from __future__ import annotations

from .types import ProviderAdapter, ProviderName


class ProviderRegistry:
    def __init__(self) -> None:
        self._providers: dict[ProviderName, ProviderAdapter] = {}

    def register(self, provider: ProviderAdapter) -> None:
        self._providers[provider.provider_name] = provider

    def get(self, name: ProviderName) -> ProviderAdapter | None:
        return self._providers.get(name)

    def list(self) -> list[ProviderAdapter]:
        return list(self._providers.values())
